### PR TITLE
Backpressure

### PR DESCRIPTION
**Pull request by nikostoulas**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "rabbit-queue",
-  "version": "4.1.2",
+  "version": "4.2.0-beta",
```
Comment (alkiskal) [inline package.json]: Is the version the intended one?

Comment (nikostoulas) [reply package.json]: Will bump the version after this is merged

File: ts/queue.ts [hunk 2]
```diff
@@ -87,23 +87,33 @@ export default class Queue {
...
+          let id = 0;
           for await (let chunk of reply) {
+            properties.correlationId = `${correlationId}.${id++}`;
             if (chunk instanceof Buffer) chunk = chunk.toString();
-            const replyBuffer = encode(chunk);
-            this.channel.sendToQueue(replyTo, replyBuffer, properties);
+            if (headers.backpressure) {
+              await Queue.getReply(chunk, properties, this.channel, replyTo, null, headers.timeout);
+            } else {
+              this.channel.sendToQueue(replyTo, encode(chunk), properties);
```
Comment (nikosd23) [inline ts/queue.ts]: Add a log here for debug

File: ts/queue.ts [hunk 3]
```diff
@@ -87,23 +87,33 @@ export default class Queue {
...
+            }
           }
           this.channel.sendToQueue(replyTo, encode(null), properties, ack);
         } catch (e) {
+          logger.info(`[${correlationId}] -> Publishing to queue ${replyTo} error ${e}`);
```
Comment (nikosd23) [inline ts/queue.ts]: Shouldn't this be logger.error?

File: ts/queue.ts (hunk 2 above)
Comment (nikostoulas) [reply ts/queue.ts]: Done

File: ts/queue.ts (hunk 3 above)
Comment (nikostoulas) [reply ts/queue.ts]: I was between info and error. Since you also think error is better I will change it.

**Pull request merged**